src/OctopusPuppet.Tests/TestHelpers/DeploymentPlannerTestFactory.cs

```diff
@@ -0,0 +1,81 @@
+﻿using NSubstitute;
+using Octopus.Client.Model;
+using Octopus.Client.Repositories;
+using Octopus.Client;
+using OctopusPuppet.DeploymentPlanner;
+using OctopusPuppet.OctopusProvider;
+using System;
+using System.Collections.Generic;
+
+namespace OctopusPuppet.Tests.TestHelpers
+{
+    public static class DeploymentPlannerTestFactory
+    {
+        public static OctopusDeploymentPlanner GetSutForVersion(string versionNumber, string environmentName = "D1")
+        {
+            var repo = Substitute.For<IOctopusRepository>();
+
+            // Projects
+            var project1 = new ProjectResource { Id = "124", Name = "ArmSharedInfrastructure", IsDisabled = false };
+            var project2 = new ProjectResource { Id = "125", Name = "Filebeat", IsDisabled = false };
+
+            repo.Projects.Get(Arg.Any<string>()).Returns(project1);
+            repo.Projects.FindAll().Returns(new List<ProjectResource> { project1, project2 });
+            repo.Projects.GetAll().Returns(new List<ReferenceDataItem>
+            {
+                new ReferenceDataItem("124",""),
+                new ReferenceDataItem("125",""),
+            });
+
+            // Releases
+            repo.Projects.GetReleases(Arg.Any<ProjectResource>())
+                .Returns(args =>
+                {
+                    var project = args.Arg<ProjectResource>();
+                    return new ResourceCollection<ReleaseResource>(
+                        new[] { new ReleaseResource(versionNumber, project.Id, "") { Id = versionNumber, Version = versionNumber } },
+                        new LinkCollection()
+                    );
+                });
+
+            // Environments
+            var envRepo = Substitute.For<IEnvironmentRepository>();
+            envRepo.GetAll().Returns(new List<ReferenceDataItem> { new ReferenceDataItem("124", environmentName) });
+            repo.Environments.Returns(envRepo);
+
+            // Dashboard
+            var dashboardRepo = Substitute.For<IDashboardRepository>();
+            dashboardRepo.GetDynamicDashboard(Arg.Any<string[]>(), Arg.Any<string[]>())
+                .Returns(new DashboardResource
+                {
+                    Projects = new List<DashboardProjectResource>
+                    {
+                        new DashboardProjectResource { Id = "124", Name = "ArmSharedInfrastructure" },
+                        new DashboardProjectResource { Id = "125", Name = "Filebeat" }                    },
+                    Items = new List<DashboardItemResource>
+                    {
+                        new DashboardItemResource { ProjectId = "124", EnvironmentId = environmentName, ReleaseVersion = versionNumber },
+                        new DashboardItemResource { ProjectId = "125", EnvironmentId = environmentName, ReleaseVersion = versionNumber },
```

Comment:
Because the repo has only two projects, and they have the same version number we're not actually able to test that we're filtering the list (i.e. returning a _partial_ list) when using `skipNoBranchSuffix`. Instead we end up testing that either all or nothing are deployed. It also means that the context of the test is lost.

I think it would be better to drive this dynamically from the test so we can really target the use case we're trying to prove.